src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    static final int INITIAL_THRESHOLD = 12;
+    private MyHashMap.Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    public MyHashMap() {
+        loadFactor = DEFAULT_LOAD_FACTOR;
+        table = new MyHashMap.Node[INITIAL_CAPACITY];
+        threshold = INITIAL_THRESHOLD;
+    }
+
+    private int hash(Object key) {
+        return key == null ? 0 : (key.hashCode() < 0 ? key.hashCode() * -1 : key.hashCode());
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        int index = newNode.hash % table.length;
```

Comment:
```suggestion
        int index = hash % table.length;
```
тут можна замінити на hashs і тоді поле .hash ноди взагалі не буде використовуватись і його можан прибрати

Reply:
Змінив

Reply:
Тут я дозволив собі піти трохи далі і замінив метод `int hash(Object key)` на `int calculateIndex(K key)`
По-перше, мінус один рядок коду у методі `put` :-)
По-друге, замінив в типі аргумента метода недолугий `Object` на кошерний дженерік `K`

Reply:
Також у методі `List<Node<K, V>> entryList()` замінив цикл `for i` на `foreach`